session.go

```diff
@@ -33,23 +33,37 @@ func RequestSkipperCallOption(requestSkipper RequestSkipper) CallOption {
 	}
 }
 
+// PrematureRefreshThresholdOption sets the threshold for a premature token
+// refresh
+func PrematureRefreshThresholdOption(accessToken, refreshToken time.Duration) CallOption {
+	return func(gcs *goCloakSession) error {
+		gcs.prematureRefreshTokenRefreshThreshold = int(refreshToken.Seconds())
+		gcs.prematureAccessTokenRefreshThreshold = int(accessToken.Seconds())
+		return nil
+	}
+}
+
 type goCloakSession struct {
-	clientID       string
-	clientSecret   string
-	realm          string
-	gocloak        gocloak.GoCloak
-	token          *gocloak.JWT
-	lastRequest    time.Time
-	skipConditions []RequestSkipper
+	clientID                              string
+	clientSecret                          string
+	realm                                 string
+	gocloak                               gocloak.GoCloak
+	token                                 *gocloak.JWT
+	lastRequest                           time.Time
+	skipConditions                        []RequestSkipper
+	prematureRefreshTokenRefreshThreshold int
+	prematureAccessTokenRefreshThreshold  int
 }
 
 // NewSession returns a new instance of a gocloak Session
 func NewSession(clientID, clientSecret, realm, uri string, calloptions ...CallOption) (GoCloakSession, error) {
 	session := &goCloakSession{
-		clientID:     clientID,
-		clientSecret: clientSecret,
-		realm:        realm,
-		gocloak:      gocloak.NewClient(uri),
+		clientID:                              clientID,
+		clientSecret:                          clientSecret,
+		realm:                                 realm,
+		gocloak:                               gocloak.NewClient(uri),
+		prematureAccessTokenRefreshThreshold:  0,
```

Comment:
Wouldn't a default value of something like 10 or 30 seconds make more sense, even if it were a breaking change?

Reply:
Sure, can do :+1: 
This would not even be a breaking change, it just would slightly alter (improve?) the behavior, which in my opinion, is acceptable when using a later version of a lib :slightly_smiling_face: 

@hanspolo or @Nerzal your opinion on setting a default >0 and if so, what value would you prefer :thinking:

Reply:
30 seconds might be too high, I would prefer 10 or even less.

Reply:
i'd also say 10 is okay.
As this behavior is configureable even 0 is okay in my opinion.